core-rust/state-manager/src/jni/transaction_preparer.rs

```diff
@@ -116,6 +116,102 @@ extern "system" fn Java_com_radixdlt_transaction_TransactionPreparer_prepareInte
     )
 }
 
+// Yes, this isn't a full transaction specification, but it's good enough for now.
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+struct PrepareTransactionIntentV2Request {
+    network_definition: NetworkDefinition,
+    header: TransactionHeaderJava,
+    subintent_count: usize,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+struct PrepareTransactionIntentV2Response {
+    raw_transaction_intent: RawTransactionIntent,
+    transaction_intent_hash: TransactionIntentHash,
+    subintent_hashes: Vec<SubintentHash>,
+}
+
+#[no_mangle]
+extern "system" fn Java_com_radixdlt_transaction_TransactionPreparer_prepareTransactionIntentV2(
```

Comment:
For future consideration, we may want to move all the test-related JNI interfaces to a dedicated folder `jni-test`.

Reply:
Yeah I think it's a nice idea. Will see if I can find time to do it now, if not we can do it in a future PR 👍